qt/applications/workbench/workbench/plugins/workspacewidget.py

```diff
@@ -137,7 +139,19 @@ def _do_show_data(self, names):
                     presenter.show_view()
                 except ValueError:
                     logger.error(
-                        "Could not open workspace: {0} with neither MatrixWorkspaceDisplay nor TableWorkspaceDisplay.")
+                        "Could not open workspace: {0} with neither "
+                        "MatrixWorkspaceDisplay nor TableWorkspaceDisplay."
+                        "".format(ws.name()))
+
+    def _do_show_algorithm_history(self, names):
+        for name in names:
+            if any(isinstance(self._ads.retrieve(name), ws_type)
+                   for ws_type in [MatrixWorkspace, ITableWorkspace]):
```

Comment:
@hsautessella @martyngigg Why was MDWorkspaces excluded?

Reply:
I've opened a PR #25119. Do let me know if there are any other options that may need adding.